tester.php

```diff
@@ -22,9 +22,23 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
-// @codingStandardsIgnoreStart
+// Require_login is not needed here.
+// phpcs:disable moodle.Files.RequireLogin.Missing
 require_once(__DIR__ . '/../../config.php');
-// @codingStandardsIgnoreEnd
+require_once(__DIR__ . '/locallib.php');
+
+// Check we are in debug mode to use this tool.
+$saml2auth = new \auth_saml2\auth();
+if (!$saml2auth->is_debugging()) {
```

Comment:
I'm not sure I want to disable this test page if debugging is off. For example I only ever turn debugging on if there is a problem I need to debug, but I use the testing page all the time when configuring it. Is there some security concern you have?

Reply:
Yep, I have slight security concerns that user may get info there she is not supposed to see. Notice it is not checking moodle DEBUG_DEVELOPER mode, it is checking if Debugging is enabled in the plugin settings only:
![image](https://user-images.githubusercontent.com/329780/119109818-69635b80-ba19-11eb-8608-43c6d810325e.png)

It is generally not good exposing test pages on production sites that are available for general public.